BE: reset dynamo before each test in test_ops_gradients.py

shunting314 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #131397
* #131372
* #131388
* #131551

https://github.com/pytorch/pytorch/pull/126586 tried to reset dynamo before each unit test. That PR get reverted a couple of times because we see post-land test failures that we don't see before merge. This PR only reset dynamo before each tests in `test_ops_gradients.py` to make it easier to land.

Eventually after we reset dynamo in each individual test files, we can move the change to the base class (TestCase) and remove the change in individual test files.

shunting314 committed:
BE: reset dynamo before each test in test_ops_gradients.py

[ghstack-poisoned]

zou3519 commented on test/test_ops_gradients.py:
```diff
@@ -36,6 +36,13 @@ def test_fn_grad(self, device, dtype, op):
             self._grad_test_helper(device, dtype, op, op.get_op())
 
+    def setUp(self):
+        TestGradients.setUp(self)
+        # reset dynamo cache to avoid issues like
+        # https://github.com/pytorch/pytorch/issues/125967#issuecomment-2118483919
+        # which depends on test order.
+        torch._dynamo.reset()
+
```
Btw, torch._dynamo.reset() might be expensive (when called so many times), so you might want to guard this on if we are doing PYTORCH_TEST_WITH_INDUCTOR

zou3519 reviewed (changes requested):
Sorry, I changed my assessment: see the comments in https://github.com/pytorch/pytorch/pull/131388

shunting314 committed:
Update on "BE: reset dynamo before each test in test_ops_gradients.py"


https://github.com/pytorch/pytorch/pull/126586 tried to reset dynamo before each unit test. That PR get reverted a couple of times because we see post-land test failures that we don't see before merge. This PR only reset dynamo before each tests in `test_ops_gradients.py` to make it easier to land.

Eventually after we reset dynamo in each individual test files, we can move the change to the base class (TestCase) and remove the change in individual test files.


[ghstack-poisoned]

shunting314 committed:
Update on "BE: reset dynamo before each test in test_ops_gradients.py"


https://github.com/pytorch/pytorch/pull/126586 tried to reset dynamo before each unit test. That PR get reverted a couple of times because we see post-land test failures that we don't see before merge. This PR only reset dynamo before each tests in `test_ops_gradients.py` to make it easier to land.

Eventually after we reset dynamo in each individual test files, we can move the change to the base class (TestCase) and remove the change in individual test files.


[ghstack-poisoned]

shunting314 commented:
@pytorchbot merge

shunting314 commented:
@pytorchbot merge

kit1980 commented:
This stack should have ciflow/inductor label.

ZainRizvi commented:
Please rebase this PR onto the latest viable/strict in order to get a functional CI signal.
